### PR TITLE
Integration testing

### PR DESCRIPTION
**Pull request by khuongTiet**
# Summary
Added better test cases for all of the routes, checking for the actual response from the call and its body through use of `assert.Equal()`

## PR - Merge Checklist
- [ ] CR'd
- [ ] README updated or N/A
- [ ] Fully tested and approved

### CODE REVIEW

File: internal/integration/bounce_rule_test.go
Comment (xdsxc) [inline internal/integration/bounce_rule_test.go]: Instead of doing this, you could use `assert.Equal(suite.T(), reqBody, br)`. That way you test the entire contents.

Style tip: When you're writing tests its nice to have variables named `have` and `want` to make it easier for the reader to determine what is being tested. Then you'll have a statement like this later: `assert.Equal(t, have, want, "something didn't match")`

File: internal/integration/bounce_rule_test.go
Comment (xdsxc) [inline internal/integration/bounce_rule_test.go]: Same comment as above re: reusing `reqbody`

File: internal/integration/change_log_test.go
Comment (xdsxc) [inline internal/integration/change_log_test.go]: There's other functions you can use for this like `assert.Len` 
https://godoc.org/github.com/stretchr/testify/assert#Len

File: internal/integration/change_log_test.go
Comment (xdsxc) [inline internal/integration/change_log_test.go:62]: General comment on this type of function: Its better to create the test data explicitly during the test run execution rather than depend on the value inside the test database. There are a few reasons for this:

- Some of the values that are pulled out look magical to someone who isn't familiar with the code. For example, its not immediately apparent where 204 comes from in `assert.Equal(suite.T(), 204, changelogs[0].ID)`.
- It ties your test case data to the in the bootstrapped data in the database.
- You can get into trouble if any of your test case modifies data in the database when a different test case tries to read it and the data is different.

I'll leave it up to you guys if you want to change it.

File: internal/sgbouncewizard/sgbouncewizard.go
Comment (xdsxc) [inline internal/sgbouncewizard/sgbouncewizard.go]:
```suggestion
```

**Review by xdsxc — DISMISSED** [review]
I think overall everything looks good. There are only a few changes that I'd like to see for this PR:
- Remove that debug line
- Validate the entire contents of responses instead of just a few fields

Like I said I'll leave the test data generation part up to you

File: internal/integration/bounce_rule_test.go
Comment (xdsxc) [inline internal/integration/bounce_rule_test.go]: These statements are backwards :(

`want` is what you're looking for, `have` is what you got. So your `want` is what you expect to receive from the server, and your `have` will be what you actually receive back

So in this case your `want` will be the thing you're adding (since thats a known value), and your `have` will be the result from retrieving from the server

Sorry, I wasn't really clear on what I meant by that

**Comment by xdsxc** [comment]
Everything looks great, just fix the mixed up variable names and we're good to go

File: internal/integration/bounce_rule_test.go
Comment (khuongTiet) [reply internal/integration/bounce_rule_test.go]: oh haha, yeah a little confusing.

I'm going to wait until Gev's PR gets merged in because I think one of his updates affects one of the tests so might as well wait for it to go through first

**Review by C0smicG — APPROVED** [review]

**Pull request merged**
